tests/assert_type/db/test_transaction.py

```diff
@@ -0,0 +1,33 @@
+from collections.abc import Callable
+from typing import List
+
+from django.db.transaction import atomic, non_atomic_requests
+from django.http import HttpRequest, HttpResponse
+
+
+@atomic
+def func1(x: int) -> List[int]: ...
+
+
+x1: Callable[[int], List[int]] = func1
```

Comment:
If due to some stubs issue, mypy thinks `@atomic` turns the function into `Any`, then this check will ignore and won't catch the issue. :(

Reply:
Crap, that's true

Reply:
It's an ugly hack, but we can fix it if we add both a positive and negative "assignment" test

```python
x1: Callable[[int], List[int]] = func1
y1: Callable[[], None] = func1  # type: ignore[...]     # <-- Expects an error to occur, but error will disappear with Any
```

Reply:
@intgr hm, couldn't the [disallow_any_decorated](https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_any_decorated) setting save us from all this happening?

Reply:
Never mind, I read that setting as if it disallowed the return type of the decorator to be `Any`. But it's even stricter; it disallows `Any` from being anywhere in the returned signature

Reply:
Instead of assigning to a different type I called the callable with wrong types e.g.

```python
x1: Callable[[int], List[int]] = func1
func1("str")
```

Which yields an error to suppress too

Reply:
(that's why I don't really like `assert_type` tests 😄)